src/containers/team-leaderboard/TeamLeaderboard.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from 'react';
+import LeaderboardTabs, {
+  TabInfo,
+} from '../../components/leaderboard/leaderboard-tabs/LeaderboardTabs';
+import { C4CState } from '../../store';
+import PageHeader from '../../components/pageheader/PageHeader';
+import styled from 'styled-components';
+import { TeamLeaderboardReducerState } from './ducks/types';
+import { connect, useDispatch } from 'react-redux';
+import { AsyncRequestKinds } from '../../utils/asyncRequest';
+import { getTeamsLeaderboard } from './ducks/thunks';
+import {
+  LEADERBOARD_DAYS,
+  LEADERBOARD_TABS,
+} from '../../components/leaderboard/constants';
+
+const LeaderboardContentContainer = styled.div`
+  padding: 100px 134px;
+`;
+
+const LeaderboardContainer = styled.div`
+  margin: 80px 0px 40px;
+`;
+
+interface TeamLeaderboardProps {
+  readonly teamLeaderboardItems: TeamLeaderboardReducerState['teamLeaderboard'];
+}
+
+const TeamLeaderboard: React.FC<TeamLeaderboardProps> = ({
+  teamLeaderboardItems,
+}) => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getTeamsLeaderboard(LEADERBOARD_DAYS.weekly));
+  }, []);
```

Comment:
Does dispatch need to be wrapped in a useEffect? Honestly not sure if it does 🤔

Reply:
Yeah it's necessary, get a maximum call stack exceeded error without it